drivers/wifi/esp8266/wifi_esp8266.c

```diff
@@ -0,0 +1,1570 @@
+	u16_t pos;
+	int bytes_left = foo_data.data_len;
+	int hdr_len;
+
+
+	sock->read_error = 0;
+	sock->rx_pkt = net_pkt_get_rx(sock->context, BUF_ALLOC_TIMEOUT);
+	if (!sock->rx_pkt) {
+		sock->read_error = -ENOMEM;
+		LOG_ERR("Failed to get net pkt");
+		goto fail;
+	}
+
+	/* set up packet data */
+	net_pkt_set_context(sock->rx_pkt, sock->context);
+	net_pkt_set_family(sock->rx_pkt, sock->family);
+
+	frag = net_pkt_get_frag(sock->rx_pkt, BUF_ALLOC_TIMEOUT);
+	if (!frag) {
+		sock->read_error = -ENOMEM;
+		net_pkt_unref(sock->rx_pkt);
+		sock->rx_pkt = NULL;
+		goto fail;
+	}
+
+	net_pkt_frag_add(sock->rx_pkt, frag);
+
+	hdr_len = net_pkt_setup_ip_data(sock->rx_pkt, sock->ip_proto,
+			&sock->src, &sock->dst);
+
+	while (inc && bytes_left) {
+		if (inc->len > bytes_left) {
+			pos = net_pkt_append(sock->rx_pkt, bytes_left,
+				inc->data, BUF_ALLOC_TIMEOUT);
+			if (pos != bytes_left) {
+				sock->read_error = -ENOMEM;
+				net_pkt_unref(sock->rx_pkt);
+				sock->rx_pkt = NULL;
+				break;
+			}
+			bytes_left = 0;
+		} else {
+			pos = net_pkt_append(sock->rx_pkt, inc->len, inc->data,
+				BUF_ALLOC_TIMEOUT);
+			if (pos != inc->len) {
+				sock->read_error = -ENOMEM;
+				net_pkt_unref(sock->rx_pkt);
+				sock->rx_pkt = NULL;
+				break;
+			}
+			bytes_left -= inc->len;
+			inc = inc->frags;
+		}
+	}
+
+	net_pkt_set_appdatalen(sock->rx_pkt, foo_data.data_len);
+	if (hdr_len > 0) {
+		frag = net_frag_get_pos(sock->rx_pkt, hdr_len, &pos);
+		NET_ASSERT(frag);
+		net_pkt_set_appdata(sock->rx_pkt, frag->data + pos);
+	} else {
+		net_pkt_set_appdata(sock->rx_pkt,
+				    sock->rx_pkt->frags->data);
+	}
+
+fail:
+	/* return data */
+	if (sock->recv_cb) {
+		if (sock->read_error) {
+			sock->recv_cb(sock->context, NULL, sock->read_error,
+				      sock->recv_user_data);
+		} else {
+			sock->recv_cb(sock->context, sock->rx_pkt, 0,
+				      sock->recv_user_data);
+		}
+	} else {
+		if (sock->rx_pkt) {
+			net_pkt_unref(sock->rx_pkt);
+		}
+	}
+
+	sock->rx_pkt = NULL;
+	*buf = net_buf_skip(*buf, foo_data.data_len);
+	foo_data.data_len = 0;
+}
+
+static int match(struct net_buf *buf, const char *str)
+{
+	struct net_buf *frag = buf;
+	int offset = 0, pos = 0;
+
+	while (frag && frag->len && *str) {
+		if (*(frag->data + offset) == *str) {
+			str++;
+			offset++;
+			pos++;
+			if (offset >= frag->len) {
+				frag = frag->frags;
+				offset = 0;
+			}
+		} else {
+			pos = -1;
+			break;
+		}
+	}
+
+	return pos;
+}
+
+static int esp8266_process_setup_read(struct net_buf **buf)
+{
+	struct net_buf *frag;
+	u8_t temp[32];
+	int hdr_len;
+	int offset;
+	char *colon;
+	char *comma;
+	int found;
+	int end;
+
+	frag = *buf;
+	end = 0;
+	offset = 0;
+	found = -1;
+	while (frag) {
+		temp[end] = *(frag->data + offset);
+		if (temp[end] == ':'){
+			found = end;
+			break;
+		}
+
+		end++;
+		offset++;
+		if (offset >= frag->len) {
+			frag = frag->frags;
+			offset = 0;
+			if (!frag) {
+				break;
+			}
+		}
+	}
+
+	if (found < 0)
```

Comment:
please place {} even around one liners, you could fix probably all of these style issues by using uncrustify

Reply:
yeah this'll get fixed with all of the rest of the style issues

Reply:
@jukkar Do you save uncrustify Zephyr config file, or point me where i could find one?

@agross-linaro If you are okay with it will try to resolve most of the formatting and minor coding issues to help this along.

Reply:
@WilliamGFish the formatting and what it is the least of the worries of getting this merged.  I wouldn't put in that kind of effort until I get around to generalizing the code to work across multiple ATA style modems.

Reply:
> @jukkar Do you save uncrustify Zephyr config file, or point me where i could find one?

Uncrustify config file can be found at zephyr root directory, the file is called `.uncrustify.cfg`